lib/models/application-invite.ts

```diff
@@ -0,0 +1,280 @@
+	 * 	if (error) throw error;
+	 * 	console.log(applicationInvite);
+	 * });
+	 */
+	exports.get = function(
+		id: number,
+		options: PineOptionsFor<ApplicationInvite> = {},
+		callback?: (error?: Error, result?: ApplicationInvite) => void,
+	): Promise<ApplicationInvite> {
+		callback = findCallback(arguments);
+		return pine
+			.get({
+				resource: RESOURCE,
+				id,
+				options: mergePineOptions(
+					{
+						$select: [
+							'id',
+							'is_invited_to__application',
+							'invitee',
+							'message',
+							'created_at',
+						],
+					},
+					options,
+				),
+			})
+			.asCallback(callback);
+	};
+
+	/**
+	 * @summary Get all Invites by application
+	 * @name getAll
+	 * @public
+	 * @function
+	 * @memberof balena.models.applicationInvite
+	 *
+	 * @description
+	 * This method returns all invites of a specific application.
+	 *
+	 * @param {number} applicationId - Application Id for w
+	 * @param {Object} [options={}] - extra pine options to use
+	 * @fulfil {Object[]} - applicationInvites
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.applicationInvite.getAllByApplication(applicationId).then(function(applicationInvites) {
+	 * 	console.log(applicationInvites);
+	 * });
+	 *
+	 * @example
+	 * balena.models.applicationInvite.getAllByApplication(applicationId, function(error, applicationInvites) {
+	 * 	if (error) throw error;
+	 * 	console.log(applicationInvites);
+	 * });
+	 */
+	exports.getAllByApplication = function(
+		applicationId: Application['id'],
+		options: PineOptionsFor<ApplicationInvite> = {},
+		callback?: (error?: Error, result?: ApplicationInvite[]) => void,
+	): Promise<ApplicationInvite[]> {
+		callback = findCallback(arguments);
+
+		return pine
+			.get({
+				resource: RESOURCE,
+				options: mergePineOptions(
+					{
+						$filter: {
+							is_invited_to__application: applicationId,
+						},
+						$expand: {
+							invitee: {
+								$select: ['id', 'email'],
+							},
+							application_membership_role: {
+								$select: ['id', 'name'],
+							},
+						},
+						$select: ['id', 'message', 'created_at'],
+					},
+					options,
+				),
+			})
+			.asCallback(callback);
+	};
+
+	/**
+	 * @summary Creates a new invite for an application
+	 * @name create
+	 * @public
+	 * @function
+	 * @memberof balena.models.applicationInvite
+	 *
+	 * @description This method invites a user by thier email/username to an application.
+	 *
+	 * @param {number} applicationId - the application id
+	 * @param {String} collaborator - the email/username of invitee
+	 * @param {String} roleTitle - the role title to be granted to the invitee
+	 * @param {String} [message=null] - the message to send along with the invite
+	 *
+	 * @fulfil {String} - application invite id
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.applicationInvite.create(applicationId,"invitee@example.org", "developer","join my app").then(function(applicationInvite) {
```

Comment:
Let's put a random appId number here as well, just so it matches the rest of the example.